pymoskito/binding_modules.py

```diff
@@ -0,0 +1,149 @@
+
+        self.add_binding_config()
+
+        # generate config
+        if os.name == 'nt':
+            result = subprocess.run(['cmake', '-A', 'x64', '.'],
+                                    cwd=self.src_path,
+                                    shell=True)
+        else:
+            result = subprocess.run(['cmake .'], cwd=self.src_path, shell=True)
+
+        if result.returncode != 0:
+            self._logger.error("Generation of binding config failed.")
+            raise BindingException("Generation of binding config failed.")
+
+    def build_binding(self):
+        # build
+        if os.name == 'nt':
+            result = subprocess.run(
+                ['cmake', '--build', '.', '--config', 'Release'],
+                cwd=self.src_path,
+                shell=True)
+        else:
+            result = subprocess.run(['make'], cwd=self.src_path, shell=True)
+
+        if result.returncode != 0:
+            self._logger.error("Build failed!")
+            raise BindingException("Build failed!")
+
+    def add_binding_config(self):
+        """
+        Add the module config to the cmake lists.
+
+        """
+        config_line = "pybind11_add_module({} {} {})".format(
+            self.module_name,
+            self.module_name + '.cpp',
+            'binding_' + self.module_name + '.cpp')
+
+        with open(self.cmake_lists_path, "r") as f:
+            if config_line in f.read():
+                return
+
+        self._logger.info("Appending build info for '{}'".format(self.module_name))
+        with open(self.cmake_lists_path, "a") as f:
+            f.write("\n")
+            f.write(config_line)
+
+    def create_cmake_lists(self):
+        """
+        Create the stub of a `CMakeLists.txt` .
+
+        Returns:
+
+        """
+        c_make_lists = "cmake_minimum_required(VERSION 2.8.12)\n"
+        c_make_lists += "project({})\n\n".format(self.module_name)
+
+        c_make_lists += "set( CMAKE_RUNTIME_OUTPUT_DIRECTORY . )\n"
+        c_make_lists += "set( CMAKE_LIBRARY_OUTPUT_DIRECTORY . )\n"
+        c_make_lists += "set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY . )\n\n"
+
+        c_make_lists += "foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )\n"
+        c_make_lists += "\tstring( TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG )\n"
+        c_make_lists += "\tset( CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} . )\n"
+        c_make_lists += "\tset( CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} . )\n"
+        c_make_lists += "\tset( CMAKE_ARCHIVE_OUTPUT_DIRECTORY_${OUTPUTCONFIG} . )\n"
+        c_make_lists += "endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )\n\n"
+
+        # TODO get pybind install via pip running and use this line:
+        # c_make_lists += "find_package(pybind11)"
+        c_make_lists += "add_subdirectory({} pybind11)\n".format(self.pybind_path)
+
+        with open(self.cmake_lists_path, "w") as f:
+            f.write(c_make_lists)
+
+    def get_module_instance(self, className):
```

Comment:
Is it planned to actually use the constructor of the bindings (which is empty for now)  instead of calling the `create` method afterwards?
If this is not the case (which would be quite sad) the call to `get_module_instance` should actually return an instance of the object instead of its class.
In my opinion, the two calls could be combined into `init_binding` which would take all arguments for the constructor, instantiate the binding object and then store it in `self.binding` or something like that.

Reply:
Yes you are right, the current situation is only for the initial used example. The create method should not used in general. There is not problem to return instance instead of the object. I will rewrite the example for this.